morepath/directive.py

```diff
@@ -210,7 +208,7 @@ def __init__(self, dispatch, name, default, index,
         self._after = after
 
     def identifier(self, predicate_registry):
-        return self.dispatch.wrapped_func, self._before, self._after
+        return self.dispatch, self._before, self._after
```

Comment:
This is presumably a consequence of you wrapping dispatch in Morepath. In the dispatchmethods-decoupled branch this can remain unchanged as the Reg dispatch object is directly available at this point.